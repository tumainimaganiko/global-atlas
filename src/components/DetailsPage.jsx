import React from 'react';
import { useLocation } from 'react-router';
import { NavLink } from 'react-router-dom';
import { FaRegArrowAltCircleRight, FaAngleLeft } from 'react-icons/fa';
import styles from '../styles/DetailsPage.module.css';

const DetailsPage = () => {
  const location = useLocation();
  const { country } = location.state;
  return (
    <div className={styles.container}>
      <NavLink className={styles.link} to="/">
        <FaAngleLeft />
      </NavLink>
      <div>
        <h2>{country.name.official}</h2>
        <p className={styles.flag}>{country.flag}</p>
        <div className={styles.details}>
          {country.capital[0] && (
            <p>
              Capital City
              <span>
                {country.capital[0]}
                {' '}
                <span>
                  <FaRegArrowAltCircleRight />
                </span>
              </span>
            </p>
          )}
          {country.continents && (
            <p>
              Continent
              <span>
                {country.continents}
                <span>
                  <FaRegArrowAltCircleRight />
                </span>
              </span>
            </p>
          )}
          {country.timezones && (
            <p>
              Timezone
              <span>
                {country.timezones[0]}
                <span>
                  <FaRegArrowAltCircleRight />
                </span>
              </span>
            </p>
          )}
          {country.population !== 0 && (
            <p>
              Population
              <span>
                {country.population}
                <span>
                  <FaRegArrowAltCircleRight />
                </span>
              </span>
            </p>
          )}
          {country.area && (
            <p>
              Area in square meters
              <span>
                {country.area}
                <span>
                  <FaRegArrowAltCircleRight />
                </span>
              </span>
            </p>
          )}
          {country.subreagion && (
            <p>
              Sub region
              <span>
                {country.subregion}
                <span>
                  <FaRegArrowAltCircleRight />
                </span>
              </span>
            </p>
          )}
        </div>
      </div>
    </div>
  );
};

export default DetailsPage;
