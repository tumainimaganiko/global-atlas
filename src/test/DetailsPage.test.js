import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';
import axios from 'axios';
import myCountries from './countryData';
import DetailsPage from '../components/DetailsPage';
import { configureStore } from '@reduxjs/toolkit';

jest.mock('axios');
const reducer = (
    state = {
        home: {home: myCountries},
    }
) => state;
const store = configureStore({reducer})

it('Should Render Country details', () => {
    axios.get.mockResolvedValue({data: myCountries})
    const tree = renderer.create(
        <MemoryRouter>
            <Provider store={store}>
                <DetailsPage />
            </Provider>
        </MemoryRouter>
    ).toJSON();
    expect(tree).toMatchSnapshot();
})