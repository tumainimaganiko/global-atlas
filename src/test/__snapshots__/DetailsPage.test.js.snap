// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should Render Country details 1`] = `
<div
  className="container"
>
  <a
    aria-current="page"
    className="link active"
    href="/"
    onClick={[Function]}
  >
    <svg
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      strokeWidth="0"
      style={
        Object {
          "color": undefined,
        }
      }
      viewBox="0 0 256 512"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M31.7 239l136-136c9.4-9.4 24.6-9.4 33.9 0l22.6 22.6c9.4 9.4 9.4 24.6 0 33.9L127.9 256l96.4 96.4c9.4 9.4 9.4 24.6 0 33.9L201.7 409c-9.4 9.4-24.6 9.4-33.9 0l-136-136c-9.5-9.4-9.5-24.6-.1-34z"
      />
    </svg>
  </a>
  <div>
    <div
      className="details"
    />
  </div>
</div>
`;
